src/fake/Roc.ts

```diff
@@ -77,12 +77,14 @@ export class FakeDocument extends BaseRocDocument<FakeRoc> {
 
     this.checkConflict();
     const doc = this.value!;
+    const owners = new Set(doc.$owner);
     if (typeof groups === 'string') {
-      doc.$owner.push(groups);
+      owners.add(groups);
     } else {
-      doc.$owner = doc.$owner.concat(groups);
+      groups.forEach(group => owners.add(group));
     }
-    return doc.$owner;
+    doc.$owner = Array.from(owners);
+    return doc.$owner.slice();
```

Comment:
it didn't copy the array before. is that intended?

Reply:
Yes, this is because there's a reference to this array in the class's state and I don't want the api user to generate a side effect on it.